### PR TITLE
LRDOCS-9472 Sample Code

### PR DESCRIPTION
**Pull request by mtokudome**
https://issues.liferay.com/browse/LRDOCS-9472

@jhinkey This sample code is based on this article: https://help.liferay.com/hc/en-us/articles/360034978832-Using-DDM-Form-Annotations-in-Configuration-Forms

To see the settings UI, navigate to System Settings > Platform > Third Party

### CODE REVIEW

File: docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationForm.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public interface V1D9ConfigurationForm {
+
+	@DDMFormField(label = "Checkbox", properties = "showAsSwitcher=true")
+	public boolean checkbox();
+
+	@DDMFormField(label = "Date", type = "date")
+	public String date();
+
+	@DDMFormField(
+		label = "Numeric", properties = "placeholder=%Seconds",
```
Comment (mtokudome) [inline docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationForm.java:44]: @jhinkey Something weird I found is that if this '%' is removed, the whole thing crashes :man-shrugging:

File: docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationDDMFormDeclaration.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.acme.v1d9.internal.configuration.admin.definition;
+
+import com.liferay.configuration.admin.definition.ConfigurationDDMFormDeclaration;
+
+import org.osgi.service.component.annotations.Component;
+
+@Component(
+	immediate = true,
```
Comment (jhinkey) [inline docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationDDMFormDeclaration.java:8]: @mtokudome Can you test removing `immediate = true` and let me know if it still works as expected?

Comment (mtokudome) [reply docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationDDMFormDeclaration.java]: Good catch, I removed the line and this still works as expected

File: docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+v1d9-configuration-name=V1D9 Configuration
+
```
[the remark this answers is not in the document]
Comment (mtokudome) [reply docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/resources/content/Language.properties]: Yes, removed trailing empty lines

File: docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationForm.java (hunk 1 above)
Comment (jhinkey) [reply docs/dxp/latest/en/developing-applications/core-frameworks/configurable-application/ddm-form-annotations/resources/liferay-v1d9.zip/v1d9-impl/src/main/java/com/acme/v1d9/internal/configuration/admin/definition/V1D9ConfigurationForm.java:44]: OK. In liferay-portal, the `%` prefix seems to be the convention. :-)

`git grep "placeholder=" $(git grep -l "@DDMFormField" *.java)`

**Comment by jhinkey** [comment]
I'm closing this until we find out more about using `%`-prefixed values for labels.

**Pull request closed without merging**